python/tskit/trees.py

```diff
@@ -4549,6 +4549,62 @@ def subset(self, nodes, record_provenance=True):
         tables.subset(nodes, record_provenance)
         return tables.tree_sequence()
 
+    def union(
+        self,
+        other,
+        node_mapping,
+        check_shared_equality=True,
+        add_populations=True,
+        record_provenance=True,
+    ):
+        """
```

Comment:
This bit (that I think I wrote?) is I think hard to understand. How about:
"
        Returns an expanded tree sequence which contains the node-wise union of
        ``self`` and ``other``, adding the non-shared portions of ``other`` onto ``self``.
        The "shared" portions are specified using a mapping
        which specifies which nodes in ``other`` are equivalent to those in ``self``: the
        ``node_mapping`` argument that should be an array of length equal to the number
        of nodes in ``other``, whose entries are the ID of the matching node in ``self``,
        or ``tskit.NULL`` if there is no matching node. the nodes that map to NULL
        will be added to ``self``, along with:
"

Reply:
nice!